api/s2n.h

```diff
@@ -2879,6 +2916,17 @@ S2N_API extern int s2n_connection_get_actual_protocol_version(struct s2n_connect
  */
 S2N_API extern int s2n_connection_get_client_hello_version(struct s2n_connection *conn);
 
+/**
+ * Access the protocol version written in the Client Hello record header.
+ *
+ * @note This field has been deprecated and is no longer used to determine TLS version.
+ * Therefore this method should only be used for logging or fingerprinting.
```

Comment:
I don't love this phrasing, but I'm struggling to improve it. I think my concern is that "no longer used to determine TLS version" implies "previously used to determine TLS version", which I think will cause more confusion between all our various version getters.

Maybe "This field has been deprecated and should not be confused with the client hello version. It is often set very low, usually to TLS1.0 for compatibility reasons, and should never be set higher than TLS1.2. Therefore this method should only be used for logging or fingerprinting."